docs/troubleshoot/troubleshoot-zos-certificate.md

```diff
@@ -51,3 +51,37 @@ ZOSMF_CERTIFICATE=
 ```
 
 4. Restart Zowe and check whether the error is resolved.
+
+## PKCS 12 server keystore generation fails in Java 8 SR7FP15, SR7 FP16, and SR7 FP20
+
+**Symptoms**
+
+When you let Zowe server installs create a PKCS12 keystore, the keystore that is generated cannot be read by ZSS. Then, parts of Zowe cannot be used. 
+
+**Solutions**
+
+This error occurs because of the incompatibility which is found between Java and GSK regarding cryptography.
+
+You can try the following options if you are effected by this error.
+
+- You can temporarily downgrade Java, for example, to Java 7.
+
+- Or you can also use the flags as below when generating a keystore. 
+
+```
+-J-Dkeystore.pkcs12.certProtectionAlgorithm=PBEWithSHAAnd40BitRC2 -J-
+Dkeystore.pkcs12.certPbeIterationCount=50000 -J-
+Dkeystore.pkcs12.keyProtectionAlgorithm=PBEWithSHAAnd3KeyTripleDES -J-
+Dkeystore.pkcs12.keyPbeIterationCount=50000
+```
+
+- Or, to create a PKCS12 keystore that can be loaded, set the system property `keystore.pkcs12.legacy` which overrides the values of the properties defined below with legacy algorithms. This system property can only be enabled and has no value. 
```

Comment:
Is it correct to say `the properties defined below` when there are no properties following? Can be a bit confusing for readers. 
One more detail, in the first example we are using the word `flags`, and in the second `system property`, could we use 1 term for both?

Reply:
Thanks for your clarification! Yes if the two terms point to one concept, we should unify them. Please see the latest commit.

Reply:
updates looks good to me, thanks